src/electron/lanceDB/lancedb.ts

```
@@ -0,0 +1,170 @@
+  async addEmbedding(id: number, filename: string, description: string, vector: Float32Array): Promise<void> {
+    await this.ensureInitialized();
+    const record: ImageEmbedding = {
+      id,
+      filename,
+      description,
+      vector,
+    };
+    await this.table.add([record]);
+  }
+
+  async searchSimilar(queryVector: Float32Array, limit: number = 3): Promise<any[]> {
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)